libs/MeshKernel/src/CurvilinearGrid/CurvilinearGrid.cpp

```diff
@@ -941,3 +959,140 @@ meshkernel::BoundingBox CurvilinearGrid::GetBoundingBox() const
 
     return BoundingBox(lowerLeft, upperRight);
 }
+
+std::vector<meshkernel::Point> CurvilinearGrid::ComputeNodes() const
+{
+    if (!IsValid())
+    {
+        throw AlgorithmError("Invalid curvilinear grid ");
+    }
+
+    std::vector<Point> result(NumN() * NumM());
+    UInt ind = 0;
+    for (UInt n = 0; n < NumN(); n++)
+    {
+        for (UInt m = 0; m < NumM(); m++)
+        {
+
+            result[ind] = m_gridNodes(n, m);
+            ind++;
+        }
+    }
+    return result;
+}
+
+std::vector<meshkernel::Edge> CurvilinearGrid::ComputeEdges() const
+{
+    if (!IsValid())
+    {
+        throw AlgorithmError("Invalid curvilinear grid ");
+    }
+
+    const auto numM = NumM();
+    const auto numN = NumN();
+
+    std::vector<Edge> result(numM * (numN - 1) +
+                             (numM - 1) * numN);
+
+    UInt ind = 0;
+    for (UInt n = 0; n < numN - 1; n++)
+    {
+        for (UInt m = 0; m < numM; m++)
+        {
+
+            result[ind].first = numM * n + m;
+            result[ind].second = numM * (n + 1) + m;
+            ind++;
+        }
+    }
+    for (UInt n = 0; n < numN; n++)
+    {
+        for (UInt m = 0; m < numM - 1; m++)
+        {
+            result[ind].first = numM * n + m;
+            result[ind].second = numM * n + m + 1;
+            ind++;
+        }
+    }
+
+    return result;
+}
+
+std::vector<meshkernel::Point> CurvilinearGrid::ComputeEdgesCenters() const
+{
+    if (!IsValid())
+    {
+        throw AlgorithmError("Invalid curvilinear grid ");
+    }
+
+    std::vector<Point> edgesCenters(GetNumEdges());
+    UInt index = 0;
+    for (UInt n = 0; n < NumN() - 1; n++)
+    {
+        for (UInt m = 0; m < NumM(); m++)
+        {
+            edgesCenters[index] = (m_gridNodes(n, m) + m_gridNodes(n + 1, m)) * 0.5;
+            index++;
+        }
+    }
+    for (UInt n = 0; n < NumN(); n++)
+    {
+        for (UInt m = 0; m < NumM() - 1; m++)
+        {
+            edgesCenters[index] = (m_gridNodes(n, m) + m_gridNodes(n, m + 1)) * 0.5;
+            index++;
+        }
+    }
+
+    return edgesCenters;
+}
+
+std::vector<CurvilinearGridNodeIndices> CurvilinearGrid::ComputeNodeIndices() const
+{
```

Comment:
as detailed in the comment above, I created a function for computing the face indices but not the inverse because it is not required for now (YAGNI principle